zk-token-sdk/src/encryption/auth_encryption.rs

```diff
@@ -157,6 +157,21 @@ impl AeKey {
     pub fn decrypt(&self, ciphertext: &AeCiphertext) -> Option<u64> {
         AuthenticatedEncryption::decrypt(self, ciphertext)
     }
+
+    pub fn to_bytes(&self) -> [u8; AE_KEY_LEN] {
+        self.0
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
```

Comment:
prefer a `TryFrom` impl. blindly throwing away errors in favor of the "easy" `Option` is almost always a bug

Reply:
re both comments: imo it should be consistent throughout the lib, so as the `ElGamalKeypair` already uses the `[from,to]_bytes` pattern it makes sense here as well.

We can change the elgamal one as well, since the zk_token_sdk feature doesn't seem to be enabled anywhere yet it shouldn't break anyone's existing code. wdyt?

Reply:
Yeah, let's add the `TryFrom` and `From` implementation for authenticated encryption for this PR and then update the `ElGamalKeypair` in a follow-up PR (if you have time of course 🙏 ).

For `ElGamalKeypair`, I don't think changing the interface will break anything downstream, but to be safe, let's implement `TryFrom` and `From` for these, update the existing `[from,to]_bytes` to call the functions from `TryFrom` and `From`, and then mark them deprecated in favor of the trait functions.

Reply:
Alrighty changed it and added a [seperate PR](https://github.com/anza-xyz/agave/pull/246) for the `ElGamalKeypair` part.

If we want this truly consistent throughout the libs there are **a lot** more spots that need to be touched though. The use of `[from,to]_bytes` is quite ubiquitous, even the `ed25519_dalek` types implement it.